### PR TITLE
[tests/functional] - Add functional test case covering SCK-SOCK migration

### PR DESCRIPTION
**Pull request by VihasMakwana**
**Description:** Add a new test case that covers no duplicates scenario during migration from SCK to SOCK 

**NOTE: This case will currently fail. It will become green once https://github.com/signalfx/splunk-otel-collector/pull/3879 gets released**

### CODE REVIEW

**Comment by rmfitzpatrick** [comment]
@atoulme, will these fixture changes break the implicit upgrade check added by https://github.com/signalfx/splunk-otel-collector-chart/pull/1051?

**Comment by jvoravong** [comment]
To get the EKS functional-tests up and running, you will have to close this PR and open up a different PR in a specific way due to recent project test updates. The EKS functional tests need to run with credentials that are only available in this repo.

- Create your feature branch in the https://github.com/signalfx/splunk-otel-collector-chart repo and not your personal fork.
- Create the PR with the feature branch in the https://github.com/signalfx/splunk-otel-collector-chart repo against the main branch.

File: functional_tests/functional_test.go [hunk 1]
```diff
@@ -249,15 +268,98 @@ func deployChartsAndApps(t *testing.T) {
...
+		jobClient := clientset.BatchV1().Jobs(job.Namespace)
+		_ = jobClient.Delete(context.Background(), job.Name, metav1.DeleteOptions{
+			GracePeriodSeconds: &waitTime,
+		})
 		uninstall := action.NewUninstall(actionConfig)
 		uninstall.IgnoreNotFound = true
 		uninstall.Wait = true
-		_, _ = uninstall.Run("sock")
-	})
+		_, _ = uninstall.Run("sck")
```
Comment (atoulme) [inline functional_tests/functional_test.go:356]: do not change this

**Comment by atoulme** [comment]
This PR has conflicts and is open from a fork, this won't work with our test checks. Going to close and reopen from this repository.
